src/java/htsjdk/samtools/SAMFileReader.java

```diff
@@ -203,6 +204,13 @@ public void close() {
     }
 
     /**
+     * If true, this reader will use asynchronous IO.
+     */
+    public void setUseAsyncIO(final boolean useAsyncIO) {
+        this.useAsyncIO = useAsyncIO;
```

Comment:
With `BAMFileReader` you added a constructor argument for this setting. Here you added a setter method instead. Why the different approaches? Can you use the same interface in both classes?